frontend/src/pages/WorkloadList/WorkloadListPage.tsx

```diff
@@ -136,12 +145,21 @@ class WorkloadListPageComponent extends FilterComponent.Component<
       })
       .catch(err => {
         if (!err.isCanceled) {
+          console.log(`error: ${err}`);
           this.handleAxiosError('Could not fetch workloads list', err);
         }
       });
   }
 
   render() {
+    const isMultiCluster = Object.keys(serverConfig.clusters || {}).length > 1;
+    const hiddenColumns = isMultiCluster ? ([] as string[]) : ['cluster'];
+    Toggles.getToggles().forEach((v, k) => {
+      if (!v) {
+        hiddenColumns.push(k);
+      }
+    });
+
```

Comment:
It's a good observation, and I can understand the desire to not handle the `Cluster` column in each user of `VirtualList`,  but actually I think it makes more sense to defer to the parents and not try and have a generic, re-usable component do special handling for a specific column.  Instead, this PR introduces a generic mechanism for hiding columns.

The code you mention above can not be removed because there is a bit more going on now in `VirtualList` to better support hidden columns, the code as it was did not really fully support column removal.   It's actually the Renderer code that should probably be removed, because imho the Renderer for the column should not actually be getting called at all if it's not multi-cluster.  It was actually this code, which I removed, that was blocking the `Cluster` column: https://github.com/kiali/kiali/pull/5953/files#diff-d9a135c62e989128a43034ac55d64a837f21fedc602dbeadb6709874fa32ce3bL110-L112.

I could actually go back and re-add special support for `Cluster`, but as I said, I think I'd prefer to let the parent components actually be in control and avoid the side-effect.

Reply:
> The code you mention above can not be removed because there is a bit more going on now in VirtualList to better support hidden columns, the code as it was did not really fully support column removal. It's actually the Renderer code that should probably be removed, because imo the Renderer for the column should not actually be getting called at all if it's not multi-cluster. It was actually this code, which I removed, that was blocking the Cluster column: https://github.com/kiali/kiali/pull/5953/files#diff-d9a135c62e989128a43034ac55d64a837f21fedc602dbeadb6709874fa32ce3bL110-L112.

Oh interesting I thought it was just the `Renderer` that was removing the column. Personally I'm fine with whatever approach is most simple and works. It's not much more work to duplicate the cluster column removal to each of the list pages.